clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -1557,6 +1557,11 @@ protected void checkIngestionProgress(Store store) throws InterruptedException {
     processIngestionException();
     maybeUnsubscribeCompletedPartitions(store);
 
+    if (emitMetrics.get()) {
```

Comment:
nit: Can we drop a comment that tells us the order of operations here?  It seems like making sure this comes before unsub is important to avoid this as a regression.

Also, is there a test we could write to this effect to prevent a regression?

Reply:
Just curious.. should we just remove all the emitMetrics variable? Do we have any use case that does not emit metrics?

Reply:
no, check where its disabled which we do to stop emission

Reply:
But then different versions of the same store will report data to the same quota usage metrics; it could be confusing. We need to move quota usage metric to versioned metrics.